curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/654210a9eda99477f5697a94.md

```diff
@@ -32,7 +32,7 @@ assert.match(code, /<button\s*class\s*=\s*('|")playlist-song-info\1\s*>\s*<span\
 You should interpolate `song.title` as the content of your `span` element.
 
 ```js
-assert.match(code, /<button\s*class\s*=\s*('|")playlist-song-info\1\s*>\s*<span\s*class\s*=\s*('|")playlist-song-title\2\s*>\$\{song\.title\}\s*<\/span>\s*<\/button>/)
+assert.match(code, /<button\s*class\s*=\s*('|")playlist-song-info\1\s*>\s*<span\s*class\s*=\s*('|")playlist-song-title\2\s*>[\s\S]*?\$\{song\.title\}[\s\S]*?<\/span>\s*<\/button>/)
```

Comment:
That might be a bit too much, allowing any kind of text before and after the `${song.title}` interpolation.
```suggestion
assert.match(code, /<button\s*class\s*=\s*('|")playlist-song-info\1\s*>\s*<span\s*class\s*=\s*('|")playlist-song-title\2\s*>\s*\$\{\s*song\.title\s*\}\s*<\/span>\s*<\/button>/)
```

Reply:
Thanks, my friend -- perhaps I need coffee but sleep would be better :)